skada/deep/_adversarial.py

```diff
@@ -370,6 +369,150 @@ def CDAN(
     return net
 
 
+class MDDLoss(BaseDALoss):
+    """Loss MDD.
+
+    This loss tries to minimize the disparity discrepancy between
+    the source and target domains. The discrepancy is estimated
+    through adversarial training of three networks: an encoder,
+    a task network and a discriminator.
+
+    See [35]_ for details.
+
+    Parameters
+    ----------
+    disc_criterion : torch criterion (class)
+        The criterion (loss) used to compute the
+        MDD loss for the discriminator.
+        If None, a CrossEntropyLoss is used.
+    gamma : float (default=4.0)
+        Margin parameter following [35]_
+
+    References
+    ----------
+    .. [35] Yuchen Zhang et. al. Bridging Theory and Algorithm
+            for Domain Adaptation. In International Conference on
+            Machine Learning, 2019.
+    """
+
+    def __init__(self, disc_criterion, gamma=4.0):
+        super().__init__()
+        if disc_criterion is None:
+            self.disc_criterion_ = torch.nn.CrossEntropyLoss()
```

Comment:
Is it not possible to set self.disc_criterion_ equal to the task criterion by default?

Reply:
@antoinedemathelin I can try to set it equal to base_criterion. Would it be ok ?

Reply:
Yes, something like `disc_criterion = copy.deepcopy(base_criterion)` to avoid using the same object for both networks

Reply:
Done